Source/WebCore/platform/graphics/FontMetrics.h

```diff
@@ -21,160 +21,165 @@
+        m_intDescent = lroundf(descent);
     }
 
-    float floatLineGap() const { return m_floatLineGap; }
+    const Markable<float, FontMetrics::MarkableTraits>& lineGap() const { return m_lineGap; }
+    int intLineGap() const { return m_intLineGap; }
     void setLineGap(float lineGap)
     {
-        m_floatLineGap = lineGap;
+        m_lineGap = lineGap;
         m_intLineGap = lroundf(lineGap);
     }
 
-    float floatLineSpacing() const { return m_floatLineSpacing; }
+    const Markable<float, FontMetrics::MarkableTraits>& lineSpacing() const { return m_lineSpacing; }
+    int intLineSpacing() const { return m_intLineSpacing; }
     void setLineSpacing(float lineSpacing)
     {
-        m_floatLineSpacing = lineSpacing;
+        m_lineSpacing = lineSpacing;
         m_intLineSpacing = lroundf(lineSpacing);
     }
 
-    float xHeight() const { return m_xHeight; }
+    const Markable<float, FontMetrics::MarkableTraits>& xHeight() const { return m_xHeight; }
     void setXHeight(float xHeight) { m_xHeight = xHeight; }
-    bool hasXHeight() const { return m_xHeight > 0; }
-    
-    bool hasCapHeight() const { return m_floatCapHeight > 0; }
-    float floatCapHeight() const { return m_floatCapHeight; }
+
+    const Markable<float, FontMetrics::MarkableTraits>& capHeight() const { return m_capHeight; }
+    int intCapHeight() const { return m_intCapHeight; }
     void setCapHeight(float capHeight)
     {
-        m_floatCapHeight = capHeight;
+        m_capHeight = capHeight;
         m_intCapHeight = lroundf(capHeight);
     }
-    
-    // Integer variants of certain metrics, used for HTML rendering.
-    int ascent(FontBaseline baselineType = AlphabeticBaseline) const
-    {
-        if (baselineType == AlphabeticBaseline)
-            return m_intAscent;
-        return height() - height() / 2;
-    }
-    
-    int descent(FontBaseline baselineType = AlphabeticBaseline) const
-    {
-        if (baselineType == AlphabeticBaseline)
-            return m_intDescent;
-        return height() / 2;
-    }
-
-    int height(FontBaseline baselineType = AlphabeticBaseline) const
-    {
-        return ascent(baselineType) + descent(baselineType);
-    }
 
-    int lineGap() const { return m_intLineGap; }
-    int lineSpacing() const { return m_intLineSpacing; }
-    
-    int capHeight() const { return m_intCapHeight; }
-    
-    bool hasIdenticalAscentDescentAndLineGap(const FontMetrics& other) const
-    {
-        return ascent() == other.ascent() && descent() == other.descent() && lineGap() == other.lineGap();
-    }
-
-    std::optional<float> zeroWidth() const { return m_zeroWidth; }
+    const Markable<float, FontMetrics::MarkableTraits>& zeroWidth() const { return m_zeroWidth; }
     void setZeroWidth(float zeroWidth) { m_zeroWidth = zeroWidth; }
 
-    float ideogramWidth() const { return m_ideogramWidth; }
+    const Markable<float, FontMetrics::MarkableTraits>& ideogramWidth() const { return m_ideogramWidth; }
```

Comment:
Returning a `Markable<float>` is simpler (and same size) as a `const Markable<float>&`

Reply:
Hm.. Such conversions seem to be allowed for class types only. My compiler complains as follows.
```
/Users/changseok/Projects/WebKit/WebKitBuild/Release/usr/local/include/wtf/Forward.h:87:49: error: type 'float' cannot be used prior to '::' because it has no
      members
template<typename T, typename Traits = typename T::MarkableTraits> class Markable;

/Users/changseok/Projects/WebKit/Source/WebCore/platform/graphics/FontMetrics.h:93:53: error: no viable conversion from returned value of type
      'const Markable<float, FontMetrics::MarkableTraits>' to function return type 'const int'
    const Markable<float>& lineGap() const { return m_lineGap; }
```

Reply:
I think @mdubet 's comment was about returning by value instead of returning a reference.

Reply:
Indeed, sorry that my comment wasn't clear.

Reply:
Something like `Markable<float> lineGap() const { return m_lineGap; }`, right? That produces the same compilation failures.

Reply:
No.. Something like:
```
Markable<float, FontMetrics::MarkableTraits> ideogramWidth() const { return m_ideogramWidth; }
```

Reply:
@mdubet Probably omitted the second template parameter to save himself some time. His comment mentions class size and returning by value instead of reference. So you can connect the dots.

Reply:
O.k. Reflected. Thanks for the clarification.